returnn/tf/engine.py

```diff
@@ -2348,64 +2400,64 @@ def extra_fetches_callback(seq_idx, seq_tag, **kwargs):
             out_cache[corpus_seq_idx] = {}
 
         # noinspection PyShadowingNames
-        for target_idx in range(num_targets):
-          if out_beam_sizes[target_idx] is None:
+        for output_layer_idx in range(num_output_layers):
+          if out_beam_sizes[output_layer_idx] is None:
             print("seq_idx: %i, seq_tag: %r, output %r: %r" % (
-              seq_idx[batch_idx], seq_tag[batch_idx], target_keys[target_idx], outputs[target_idx][batch_idx]),
-                  file=log.v4)
+              seq_idx[batch_idx], seq_tag[batch_idx],
+              output_layer_names[output_layer_idx], outputs[output_layer_idx][batch_idx]), file=log.v4)
             out_idx = batch_idx
           else:
             print("seq_idx: %i, seq_tag: %r, outputs %r: %r" % (
-              seq_idx[batch_idx], seq_tag[batch_idx], output_layer_names[target_idx],
-              outputs[target_idx][batch_idx * out_beam_sizes[target_idx]:(batch_idx + 1)*out_beam_sizes[target_idx]]),
+              seq_idx[batch_idx], seq_tag[batch_idx], output_layer_names[output_layer_idx],
+              outputs[output_layer_idx]
+              [batch_idx * out_beam_sizes[output_layer_idx]:(batch_idx + 1)*out_beam_sizes[output_layer_idx]]),
```

Comment:
This is hard to read. Maybe calculate the indices in separate lines. Also, spacing around `*` is inconsistent (which is was before).